crates/interpreter/src/instructions/contract.rs

```diff
@@ -426,6 +426,38 @@ pub fn extstaticcall<WIRE: InterpreterTypes, H: Host + ?Sized>(
     );
 }
 
+pub fn gas2eth<WIRE: InterpreterTypes, H: Host + ?Sized>(
+    interpreter: &mut Interpreter<WIRE>,
+    host: &mut H,
+) {
+    popn_top!([addr], gas_amount, interpreter);
+    let addr = addr.into_address();
+    let gas_price = host.effective_gas_price();
```

Comment:
Effective gas price contains `base_fee+priority_fee`, `base_fee` should be burned so we should use `priority_fee`  be removing `host.base_fee()` from it

Reply:
oh, I see. Do you think this should be clarified in the EIP?

Reply:
EIP should clarify it, as it is important and maybe even breaking the usage of GAS2ETH.

Imagine only taking into account a priority fee, this would burn more eth than transferred it, and in the end you need to know how much fee got minted/transfered so this is not added as reward to the block builder/beneficiary.

And if we want to somehow skip burning of this basefee this effect eip1559 fee mechanism.

This EIP does not feel flushed enough.

Reply:
i've proposed some changes to the EIP which fix the burn issue: https://github.com/ethereum/EIPs/pull/10129